honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,311 @@
+            mesh_pts = [to_point3d(mesh.Vertices[i])
+                        for i in range(len(mesh.Vertices))]
+        except AttributeError:
+            warnings.warn(
+                'Non-planar Breps are not handled by Honeybee. They are ignored.'
+            )
+            return faces
+        else:
+            # In the list of the Polylines, if there's only one polyline then
+            # the face has no holes
+            if len(sorted_polylines) == 1:
+                if len(mesh_pts) == 4 or len(mesh_pts) == 3:
+                    faces.extend(mesh_to_face3d(mesh))
+                else:
+                    boundary_pts = remove_dup_vertices(
+                        sorted_polylines[0].vertices, tolerance)
+                    lb_face = Face3D(boundary=boundary_pts)
+                    faces.append(lb_face)
+
+            # In the list of the Polylines, if there's more than one polyline then
+            # the face has hole / holes
+            elif len(sorted_polylines) > 1:
+                boundary_pts = remove_dup_vertices(
+                    sorted_polylines[0].vertices, tolerance)
+                hole_pts = [remove_dup_vertices(polyline.vertices, tolerance)
+                            for polyline in sorted_polylines[1:]]
+                # Merging lists of hole_pts
+                total_hole_pts = [
+                    pts for pts_lst in hole_pts for pts in pts_lst]
+                hole_pts_on_boundary = [
+                    pts for pts in total_hole_pts if pts in boundary_pts]
+                # * Check 02 - If any of the hole is touching the boundary of the face
+                # then mesh it
+                if len(hole_pts_on_boundary) > 0:
+                    warnings.warn(
+                        "The surface has holes that touch the boundary. \
+                            It will be meshed")
+                    faces.extend(brep_to_face3d(brep))
+                else:
+                    lb_face = Face3D(boundary=boundary_pts, holes=hole_pts)
+                    faces.append(lb_face)
+
+    return faces
+
+
+def solid_to_face3d(brep):
+    """Create a Ladybug Face3D object from a rhino3dm Brep.
+
+    This function is used in translating rhino3dm solid volumes to Ladybug Face3D
+    objects.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+
+    return faces
+
+
+def extrusion_to_face3d(extrusion):
+    """Create a Ladybug Face3D object from a rhino3dm Extrusion.
+
+    Args:
+        brep: A rhino3dm Extrusion.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    mesh = extrusion.GetMesh(rhino3dm.MeshType.Any)
+    faces.extend(mesh_to_face3d(mesh))
+
+    return faces
+
+
+def to_face3d(obj, *, tolerance, raise_exception=True):
```

Comment:
What is this second input argument here? I don't see it in the input Args.

Reply:
The bare `*` forces the use of named arguments after it. It's for good practice to decrease the chance of making mistake because of the order of inputs. See here: https://stackoverflow.com/questions/14301967/bare-asterisk-in-function-arguments

Calling this function as `to_face3d(obj, 0.1)` will throw an error. The arguments after * should be called by name: `to_face3d(obj, tolerance=0.1)`

Reply:
Interesting. Thank you for explaining, @mostaphaRoudsari . That's pretty nifty.